_posts/2024-01-11-gsoc2023_marinegor.md

```diff
@@ -0,0 +1,170 @@
+
+
+## How can you use it?
+
+Let's imagine we've just learned that MDAnalysis now supports parallelization, and want to calculate RMSD of our large trajectory faster using multiple cores on our machine. Imagine we have a 16-core CPU Linux workstation with an SSD drive and a 1 us trajectory of lysozyme in water. Like this:
+
+
+```python
+import MDAnalysis as mda
+from MDAnalysis.analysis import rms
+
+prefix = "./large_data/"
+traj, top = f"{prefix}/md_0_1.xtc", f"{prefix}/md_0_1.gro"
+
+u = mda.Universe(top, traj)
+```
+
+First we want to get a reference structure by taking the average of all frames. Like this:
+
+```python
+from MDAnalysis.analysis.align import AverageStructure
+
+avg = AverageStructure(mobile=u).run(backend='multiprocessing', n_workers=16)
+```
+
+but we get this:
+
+```python
+ValueError                                Traceback (most recent call last)
+Cell In[11], line 2
+      1 from MDAnalysis.analysis.align import AverageStructure
+      2 avg = AverageStructure(mobile=u).run(backend='multiprocessing', n_workers=16)
+...
+ValueError: backend=multiprocessing is not in self.available_backends=('local',) for class AverageStructure
+```
+
+which basically says we can use only `backend='local'` for the `AverageStructure`. Ok, let's do that, but with a large step to save time:
+
+
+```python
+avg = AverageStructure(mobile=u).run(step=100)
+ref = avg.results.universe
+```
+
+and start our analysis run -- RMSD for multiple selections to later compare them between each other:
+
+
+```python
+groupselections = ("protein", "backbone", "name CA")
+
+R = rms.RMSD(
+    u,  # universe to align
+    ref,  # reference universe or atomgroup
+    groupselections=groupselections,
+    select="backbone",  # group to superimpose and calculate RMSD
+)
+```
+
+If we start it with `R.run()`, we won't even know when the run would finish. Luckily, we can add some verbosity with `R.run(verbose=True)` and see a nice `tqdm` progressbar that shows us that ETA of the whole analysis is around 4 minutes:
+
+```python
+>>> R.run(verbose=True)
+5%|▌         | 5062/100001 [00:13<04:15, 371.20it/s]
+```
+
+Let's try to speed it up now. Which backends do we have available?
+
+```python
+>>> rms.RMSD.available_backends
+('local', 'multiprocessing', 'dask')
+```
+
+Let's try a built-in `multiprocessing` first:
+
+```python
+>>> R.run(backend='multiprocessing', n_workers=4)
+# CPU times: user 153 ms, sys: 74.2 ms, total: 227 ms
+# Wall time: 1min 14s
+```
+
+OK, this is roughly 4 times faster! Amazing, roughly as we expected.
+Spoiler though -- if we do it with 16 workers, we'll see the total time around 40 seconds, so improvement saturates at some point.
+
+But, we've lost something valuable when switching to `multiprocessing` -- we don't have a decent progressbar anymore. Luckily, we can use an amazing `dask` dashboard that allows us to monitor all tasks given to a particular `dask.distributed` cluster!
```

Comment:
@marinegor - I just noticed that dask.distributed is still mentioned here and L130, although it was removed from other places. Should it remain here? If not, please tell me exactly what needs to be removed/replaced, or open a PR to make the necessary change. Thanks!